### PR TITLE
Liveness probe

### LINKED ISSUE #1201: Task: Add liveness/readiness probes
The initial health checking probe was removed in #1198 to avoid issues where the Kubelet could not connect to the health endpoint on some clusters. The initial check was only doing a naive check anyway. We should regroup and investigate what these endpoints should actually do in order to ensure the operator is functioning and the reintroduce the code changes.

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #1201 

**What this PR does / why we need it**:

This PR adds liveness probe for ASOv2 which makes sure ASO is started successfully, then only pod shows `Ready` state or else goes into `Error` state.

### CODE REVIEW

File: v2/config/manager/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -22,5 +22,6 @@ spec:
       - name: manager
         args:
         - "--metrics-addr=127.0.0.1:8080"
+        - "--health-addr=:8081"
```
Comment (matthchr) [inline v2/config/manager/manager_auth_proxy_patch.yaml:25]: Does this addr need to be exposed in the service too? I assume no but just checking

**Review by matthchr — APPROVED** [review]

File: v2/config/manager/manager_auth_proxy_patch.yaml (hunk 1 above)
Comment (super-harsh) [reply v2/config/manager/manager_auth_proxy_patch.yaml:25]: Nope! it gets reached through the container port

**Review by theunrepentantgeek — COMMENTED** [review]
Looks good, but one question: How long does it wait for the health probe to succeed before signaling an error?

**Comment by theunrepentantgeek** [comment]
Those defaults seem reasonable. 😸

**Pull request merged**